matrix.py

```diff
@@ -0,0 +1,156 @@
+"""Matrix exercise, Daniel Prince - unity training."""
+
+
+class Matrix:
+    """Perform all sorts of matrix operations."""
+
+    def __init__(self, data):
+        self.data = data
+        # Check if n x n
+        for tup in self.data:
+            if len(self.data) != len(tup):
+                raise ValueError("not a valid matrix. (not nXn)")
+
+    def __str__(self):
+        return str(self.data)
+
+    def __repr__(self):
+        return '{self.__class__.__name__}({self.data})'.format(self=self)
+
+    def __iter__(self):
+        return iter(self.data)
+
+    def __hash__(self):
+        return hash(self.data)
+
+    def __eq__(self, other):
+        return self.data == other.data
+
+    def __ne__(self, other):
+        return self.data != other.data
+
+    def __add__(self, other):
+        """Adds 2 matrices."""
+
+        if isinstance(other, Matrix):
+            if len(self.data) == len(other.data):
+                post_add_list = [self.data[row_i][column_i]
+                                 + other.data[row_i][column_i]
+                                 for row_i in range(len(self.data))
+                                 for column_i in range(len(other.data))]
+                return Matrix(tuple(splitting(post_add_list, len(self.data))))
```

Comment:
As a general comment,
Instead of creating a list and converting it to tuple, create it as a tuple from the beginning.
remove splitting method its unnecessary.
Learn how to create 2 dim tuples with list comprehension.

Reply:
will do!
although in my mul function, the best way I found was by splitting a list, so I decided 
to use it to convert to tuple as well. instead of writing the same code a bunch of times.